### PR TITLE
Rename system.processes.* namespace to system.process.*

### PR DESCRIPTION
**Pull request by ChrsMark**
Fixes https://github.com/open-telemetry/semantic-conventions/issues/266

## Changes

This PR renames `system.processes.*` namespace to `system.process.*`. This is based on https://github.com/open-telemetry/semantic-conventions/pull/267 to align with `"Metric namespaces should not be pluralized"` rule.

Also it changes `system.processes.created` to `system.process.created_total` since `.process.created` would not indicate the total number of processes but possible when a process was created.

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] [CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

**Review by mx-psi — COMMENTED** [review]
LGTM but needs a few updates to make CI pass

**Review by mx-psi — APPROVED** [review]

File: model/metrics/system-metrics.yaml
Comment (joaopgrassi) [inline model/metrics/system-metrics.yaml:508]: Hum I think this goes against this: [do not use _total](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md#do-not-use-total)

File: model/metrics/system-metrics.yaml
Comment (joaopgrassi) [inline model/metrics/system-metrics.yaml:499]: Hum.. we may have a problem. I read again the metric guidelines we have today in https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md#use-count-instead-of-pluralization-for-updowncounters and it says:

>If the value being recorded represents the count of concepts signified by the namespace then the metric should be named count (within its namespace). The pluralization rule does not apply in this case.

> For example if we have a namespace system.processes which contains all metrics related to the processes then to represent the count of the processes we can have a metric named system.processes.count. The suffix count here indicates that it is the count of system.processes.

So the `.count` one seems to be correct under `processes`. But I wonder about its attributes.. and also the `.created` one. Clearly one can't be in a namespace and the other in another. 🤔

Comment (trask) [reply model/metrics/system-metrics.yaml:499]: I believe this confusion was introduced by my #267. I sent #488 to fix it.

Comment (ChrsMark) [reply model/metrics/system-metrics.yaml:499]: So @trask is the renaming to `system.process.count` still valid? As https://github.com/open-telemetry/semantic-conventions/pull/488#discussion_r1381186138 indicates the example which still uses `system.processes` does not help to clarify the case 🤔

Comment (trask) [reply model/metrics/system-metrics.yaml:499]: I fixed #488, let me know if it's still unclear, thx!

Comment (ChrsMark) [reply model/metrics/system-metrics.yaml:499]: Thank's @trask that's clear now.
@joaopgrassi if you agree feel free to resolve this one.

File: model/metrics/system-metrics.yaml
Comment (ChrsMark) [reply model/metrics/system-metrics.yaml:508]: Interesting point! I wonder though what could be a better naming here since `system.process.created` would indicate the creation time of a process maybe?

Also by `brief: "Total number of processes created over uptime of the host"` and `instrument: counter` I understand that we have to do with monotonically increasing counter but having a delta metric instead of a cumulative value would be confusing so I wonder what is the rational of [do not use _total](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md#do-not-use-total) and if it's applicable here. (Sorry if I miss anything here)

Comment (ChrsMark) [reply model/metrics/system-metrics.yaml:508]: @open-telemetry/semconv-system-approvers any thoughts on this?

Comment (joaopgrassi) [reply model/metrics/system-metrics.yaml:508]: `_total` is related to cumulative counters, so if I export to a back-end which works with delta only (by setting up my SDK to export deltas) then it is confusing, that's the reason in the guidelines:

> Counters SHOULD NOT append _total either because then their meaning will be confusing in delta backends.

If I got it right, this metric is measuring the number of process created (also spawn/initiated) since the host has started? Maybe one of those names can be used instead?

Comment (ChrsMark) [reply model/metrics/system-metrics.yaml:508]: Thank you @joaopgrassi! I would suggest the following options then:

a) `system.process.created.count`
b) `system.process.spawn`
c) `system.process.num_created`

@open-telemetry/semconv-system-approvers could you provide your feedback/preferences on which one we should go with?

Comment (trask) [reply model/metrics/system-metrics.yaml]: I like `system.process.started`. I also like `system.process.created`.

> since `system.process.created` would indicate the creation time of a process maybe?

I don't personally see this as a problem, since it's a metric name (as opposed to an attribute name).

In case it helps, we have similar scenario in JVM metrics with:

* [`jvm.class.count`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/runtime/jvm-metrics.md#metric-jvmclasscount) - current number of classes
* [`jvm.class.loaded`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/runtime/jvm-metrics.md#metric-jvmclassloaded) - number of classes loaded since start
* [`jvm.class.unloaded`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/runtime/jvm-metrics.md#metric-jvmclassunloaded) - number of classes unloaded since start

Comment (joaopgrassi) [reply model/metrics/system-metrics.yaml]: > > since `system.process.created` would indicate the creation time of a process maybe?
> 
> I don't personally see this as a problem, since it's a metric name (as opposed to an attribute name).

Yeah I also agree, I think `system.process.created` as a metric to me would indicate the number of created process over time. I think dropping the `.count` on it makes more sense.

Comment (ChrsMark) [reply model/metrics/system-metrics.yaml]: Thank's folk! I don't have really strong preference here so I agree we can go with removing the `.count` suffix. I have pushed this change with [141823e](https://github.com/open-telemetry/semantic-conventions/pull/484/commits/141823e00dd5f9e6a1dc869f8d47c3ce8a595e35).

File: CHANGELOG.md
Comment (joaopgrassi) [inline CHANGELOG.md:97]: I guess this can already be inferred by the entry above, about renaming the entire namespace?

File: model/metrics/system-metrics.yaml
Comment (joaopgrassi) [reply model/metrics/system-metrics.yaml]: I think we are good now! Resolving this

**Review by pyohannes — APPROVED** [review]

**Comment by joaopgrassi** [comment]
@open-telemetry/semconv-system-approvers please take another look

File: CHANGELOG.md
Comment (ChrsMark) [reply CHANGELOG.md]: Good catch, thank's!

**Review by mx-psi — APPROVED** [review]

**Review by AlexanderWert — APPROVED** [review]

**Review by mx-psi — DISMISSED** [review]
Blocking to avoid merge since we want to discuss this with semconv after the WG meeting yesterday

**Comment by mx-psi** [comment]
One of the points of discussion was the fact that the hostmetrics receiver has different scrapers `process` and `processes` https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/hostmetricsreceiver#getting-started and that this rename causes the metrics to have a namespace that does not match its scraper. Some previous discussion about the rationale for the namespace can be found at https://github.com/open-telemetry/semantic-conventions/issues/212#issuecomment-1685722874

**Comment by ChrsMark** [comment]
@trask any thoughts on https://github.com/open-telemetry/semantic-conventions/pull/484#issuecomment-1838335998?

**Comment by ChrsMark** [comment]
Trying to summarize our options here (outcome from discussions on last week's System WG meeting):

Option 1: Reconsider the pluralization rule for namespaces based on the inconsistency with Otel collector implantation mentioned at https://github.com/open-telemetry/semantic-conventions/pull/484#issuecomment-1838335998. Feedback from @trask 
on this would be really valuable.

Option 2: Change Otel Collector's implamentation so as conventionally to be aligned with the namespace pluralization rule of SemConv. cc: @braydonk @dmitryax 

Option 3: Keep the pluralization rule as is as well as the current Collector's implementation but add a note in the rule and the collector to clarify that `system.process.*` is for aggregated metrics while `process.*` is for metrics of a single resource.

**Comment by mlunadia** [comment]
>this rename causes the metrics to have a namespace that does not match its scraper

Option 1. Whilst I understand it would be nice to have consistency with the Otel collector implementation, enforcing this results in surfacing unnecessary implementation details to users. 

IIUC the push for consistency of scraper need to match the namespace would apply across. There are areas where this would result in namespaces that many users would not necessarily be familiar with eg. kubelestats received: namespace would need to be `k8s.kublet.*` or `k8s.kubeletstats.*` (?) to match the relevant [scraper](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/kubeletstatsreceiver/scraper.go) when these are metrics related to node, pod, container, and volume.

Options 2&3. I don't have strong opinions on either of these two options because they are more focused on the implementation without affecting the user experience. 

@jsuereth this comment is in addition to the reasoning of CPU not being plural as discussed at the last semcon SIG meeting.

**Comment by trask** [comment]
> @trask any thoughts on [#484 (comment)](https://github.com/open-telemetry/semantic-conventions/pull/484#issuecomment-1838335998)?

the trouble seems to stem from having `process.*` ([Process Metrics](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/process-metrics.md#semantic-conventions-for-os-process-metrics)) and `system.process.*` ([Aggregate System Process Metrics](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/system-metrics.md#aggregate-system-process-metrics))

what do you think of naming the scrapers

* `process` - capturing aggregated `system.process.*`, since this matches the naming for all the others, e.g. `network`, `cpu`, etc
* `process_detail` - capturing `process.*` metrics (per-process metrics)

**Comment by ChrsMark** [comment]
From the discussion on Thursday's [System WG meeting](https://docs.google.com/document/d/1p5TH57t43XpxA48onLzX4PIr3g6ydYKCtR_AUlsCnQk/edit#heading=h.3b9sz8nx597n)(Dec 14th):

We are good to go with this PR. Collector(`hostmetricsreceiver`)'s maintainers will figure out the implementation details but from SemConv point of view there is no reason to block this change/PR.

cc: @dmitryax @braydonk @jsuereth

**Comment by braydonk** [comment]
> maintainers will figure out the implementation details

Yes, I won't get into too much detail but to clarify I am planning to merge the `process` and `processes` scraper, such that just the `process` scraper produces both `process` and `system.process` metrics.

**Review by reyang — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Comment by jsuereth** [comment]
Can you rebase your CHANGELOG / Schema-next changes?  They appear to have moved into the release :)

**Pull request merged**
